src/StackViews.jl

```diff
@@ -119,7 +119,7 @@ function Base.axes(A::StackView{T,N,D}) where {T,N,D}
     prev, post = Base.IteratorsMD.split(frame_axes, Val(D-1))
 
     # use homogenous range to make _append_tuple happy
```

Comment:
I think I use `typeof(first(frame_axes))` rather than `eltype(prev)` for a "performance" reason.
Mostly `Any[...]` vs [...]`, can you check if they still work similarly in terms of performance for non-trivial large arrays (ttfx time and average time)?

Reply:
I can do some comparisons today. Used `eltype(prev)` here because `_apend_tuple` assumes a homogeneous tuple `prev` anyways, and I thought `eltype` would be something like `eltype(::NTuple{N, T}) where {N, T}=T`. Upon closer inspection, `eltype` on tuple (or rather tuple types) is a bit more complicated, but there does not seem to be obvious problems with type stability.
With previous `typeof(first(frame_axes))` there were issues because `frame_axes == ()` for 0-dim slices. Alternative: new helper `_fill_range_type` with methods for  non-empty tuples (falling back to `typeof(first(frame_axes))` and for empty tuples (returning `Base.Bottom` e.g. )